PopPUNK/lineages.py

```diff
@@ -0,0 +1,470 @@
+        kmers, sketch_sizes, codon_phased, max_search_depth, rank_list, use_accessory, min_count, \
+        count_unique_distances, reciprocal_only, strand_preserved, core, accessory, lineage_dbs = \
+          pickle.load(pickle_file)
+
+    dbFuncs = setupDBFuncs(args)
+
+    # Define clustering files
+    previous_clustering_file = os.path.join(model_dir,os.path.basename(model_dir) + '_clusters.csv')
+    external_clustering = None
+    if clustering_file != previous_clustering_file:
+        external_clustering = clustering_file
+
+    # Store clustering
+    overall_lineage = {}
+
+    # Ignore QC at the moment
+    qc_dict = {'run_qc': False, 'type_isolate': None }
+
+    # Check output file
+    if args.output is None:
+        sys.stderr.write("Need an output file name\n")
+        sys.exit(1)
+
+    # Set up database
+    createDatabaseDir = dbFuncs['createDatabaseDir']
+    constructDatabase = dbFuncs['constructDatabase']
+    readDBParams = dbFuncs['readDBParams']
+
+    if ref_db == args.output:
+        sys.stderr.write("--output and --ref-db must be different to "
+                         "prevent overwrite.\n")
+        sys.exit(1)
+
+    # construct database
+    createDatabaseDir(args.output, kmers)
+    qNames = constructDatabase(args.query,
+                                kmers,
+                                sketch_sizes,
+                                args.output,
+                                args.threads,
+                                True, # overwrite - probably OK?
+                                codon_phased = codon_phased,
+                                calc_random = False,
+                                use_gpu = args.gpu_sketch,
+                                deviceid = args.deviceid)
+
+    if qc_dict["run_qc"]:
+        pass_assembly_qc, fail_assembly_qc = \
+                sketchlibAssemblyQC(output,
+                                    qNames,
+                                    qc_dict)
+        if len(fail_assembly_qc) > 0:
+            sys.stderr.write(f"{len(fail_assembly_qc)} samples failed:\n"
+                             f"{','.join(fail_assembly_qc.keys())}\n")
+            qNames = [x for x in qNames if x in pass_assembly_qc]
+            if len(qNames) == 0:
+                sys.exit(1)
+
+    isolateClustering = \
+        assign_query_hdf5(dbFuncs,
+                    ref_db,
+                    qNames,
+                    args.output,
+                    qc_dict,
+                    False, # update DB - not yet
+                    False, # write references - need to consider whether to support ref-only databases for assignment
+                    distances,
+                    False, # serial - needs to be supported for web version?
+                    args.threads,
+                    True, # overwrite - probably OK?
+                    False, # plot_fit - turn off for now
+                    False, # graph weights - might be helpful for MSTs not for strains
```

Comment:
I would suggeest either using kwargs, or doing something like `update_db = False` then setting `update_db=update_db` in the function call. That will make this a lot clearer

Reply:
Not yet implemented database updating - worth discussing with @muppi1993 with regard to beebop functioning.

Reply:
sure, but I mean replace all the `True`/`False` arguments with a variable that's more readable